tardis/apps/publication_workflow/static/publication-form/publication_form.js

```diff
@@ -0,0 +1,504 @@
+// TODO: remove hard coded URLs in case urls.py changes
+
+// A filter used to remove items from the list of available datasets
+// when added to the list of datasets to include in the publication
+/*eslint no-unused-vars: ["error", { "argsIgnorePattern": "^_" }]*/
+angular
+.module('publicationWorkflow')
+.filter('removeMatchingDatasets', function () {
+    return function (input, compareTo) {
+        if (angular.isUndefined(input)) {
+            return input;
+        }
+        var output = [];
+        for (var i = 0; i < input.length; i++) {
+            var add = true;
+            for (var j = 0; j < compareTo.length; j++) {
+                if (input[i].id === compareTo[j].dataset.id) {
+                    add = false;
+                }
+            }
+            if (add) {
+                output.push(input[i]);
+            }
+        }
+        return output;
+    };
+});
+
+// The following two tardisForm directives attach appropriate ng-model
+// attributes to form fields
+// This code was adapted from http://stackoverflow.com/questions/21943242/child-input-directive-needs-to-compile-in-the-scope-of-its-parent-for-ng-model-t
+angular
+.module('publicationWorkflow')
+.directive('tardisForm', function () {
+    return {
+        replace: true,
+        transclude: true,
+        scope: {
+            model: '=myModel',
+            schema: '='
+        },
+        template: '<div ng-form novalidate><div ng-transclude></div></div>',
+        controller: function ($scope, $element, $attrs) {
+            var vm = this;
+            if (angular.isUndefined($scope.model)) {
+                $scope.model = {};
+            }
+            $scope.model.schema = $scope.schema;
+            vm.getModel = function () {
```

Comment:
Codacy has a fix for the issue: [Expected indentation of 16 spaces but found 12.](https://app.codacy.com/gh/mytardis/mytardis/pullRequest?prid=7933947)
```suggestion
                vm.getModel = function () {
```